unixfs/archive/tar/writer.go

```diff
@@ -19,18 +20,29 @@ import (
 // unixfs merkledag nodes as a tar archive format.
 // It wraps any io.Writer.
 type Writer struct {
-	Dag  mdag.DAGService
-	TarW *tar.Writer
+	Dag            mdag.DAGService
+	TarW           *tar.Writer
+	progressBarOut io.Writer
+	progressBar    *progressbar.ProgressBar
 
 	ctx cxt.Context
 }
 
 // NewWriter wraps given io.Writer.
-func NewWriter(ctx cxt.Context, dag mdag.DAGService, archive bool, compression int, w io.Writer) (*Writer, error) {
+func NewWriter(ctx cxt.Context, dag mdag.DAGService, archive bool, compression int, w io.Writer, progressBarOut io.Writer, size uint64) (*Writer, error) {
```

Comment:
No, this is not the right way to do this. the prograss bar stuff does not belong here. All this stuff is already complicated, and this makes it way worse. there must be _some_ way to report out the information from the Writer.

The simplicity and cleanliness of Interfaces really matters to their usability.